datasource.go

```diff
@@ -118,3 +128,75 @@ func (ds *sqldatasource) CheckHealth(ctx context.Context, req *backend.CheckHeal
 		Message: "Data source is working",
 	}, nil
 }
+
+func handleError(rw http.ResponseWriter, err error) {
+	rw.WriteHeader(http.StatusBadRequest)
+	_, err = rw.Write([]byte(err.Error()))
+	if err != nil {
+		backend.Logger.Error(err.Error())
+	}
+}
+
+type columnRequest struct {
+	Table string `json:"table"`
+}
+
+func (ds *sqldatasource) handle(resource string) func(rw http.ResponseWriter, req *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		if ds.Completable == nil {
+			handleError(rw, errors.New("not implemented"))
+			return
+		}
+
+		var res []string
+		var err error
+		switch resource {
+		case "tables":
+			res, err = ds.Completable.Tables(req.Context(), ds.db)
+		case "schemas":
+			res, err = ds.Completable.Schemas(req.Context(), ds.db)
+		case "columns":
+			if req.Body == nil {
+				handleError(rw, errors.New("missing table name in request body"))
+				return
+			}
+			reqBody, err := ioutil.ReadAll(req.Body)
+			if err != nil {
+				handleError(rw, err)
+				return
+			}
+			column := columnRequest{}
+			err = json.Unmarshal(reqBody, &column)
+			if err != nil {
+				handleError(rw, err)
+				return
+			}
+			res, err = ds.Completable.Columns(req.Context(), ds.db, column.Table)
+			if err != nil {
+				handleError(rw, err)
+				return
+			}
+		}
+		if err != nil {
+			handleError(rw, err)
+			return
+		}
+
+		resJSON, err := json.Marshal(res)
+		if err != nil {
+			handleError(rw, err)
+			return
+		}
+		rw.Header().Add("Content-Type", "application/json")
+		_, err = rw.Write(resJSON)
+		if err != nil {
+			backend.Logger.Error(err.Error())
+		}
```

Comment:
This can be reduced in the same way :)
```suggestion
		rw.Header().Add("Content-Type", "application/json")
		if err := json.NewEncoder(rw).Encode(res)
			handleError(rw, err)
			return
		}
```